Devnot.Mentor.Api/Repositories/MenteeRepository.cs

```diff
@@ -15,6 +17,40 @@ public MenteeRepository(MentorDBContext context) : base(context)
 
         }
 
+        public Task<List<Mentee>> SearchAsync(SearchRequest request)
+        {
+            if (request is null || request.IsNotValid())
+            {
+                return DbContext.Mentee.Include(mentee => mentee.User).ToListAsync();
```

Comment:
Request boş veya valid değilse, boş bir değer dönmemiz daha iyi olur. Aramaya sonuç vermek için, en az bir arama kriterinin (FullName, Title) dolu olmasını istemeliyiz.

Hatta durum buysa, Repository katmanına gitmeden Servis veya direkt olarak Controller tarafında bunu dönmemiz daha iyi olur.

Reply:
Bunu düşünme nedenim uygulama içerisinde mentor ya da mentee kısmının gösterildiği sayfa da olacak `/mentors` `/mentees` gibi. Hiçbir filtre gelmediyse direkt kayıtları verecek, filtre geldikçe de işlem yapacak. Tıpkı coding coach gibi: 

![image](https://user-images.githubusercontent.com/43035417/128521501-64dc229b-6391-4453-a649-fba5399a31d0.png)

Reply:
Tamamdır öyleyse, bir sakıncası yok. O end-point'i sadece search için kullanacağımızı düşünmüştüm.